client/client.go

```diff
@@ -24,21 +24,21 @@ type Client interface {
 	CommitAsync() *ReqRes
 
 	FlushSync() error
-	EchoSync(msg string) (res types.Result)
-	InfoSync(types.RequestInfo) (resInfo types.ResponseInfo, err error)
-	SetOptionSync(key string, value string) (res types.Result)
-	DeliverTxSync(tx []byte) (res types.Result)
-	CheckTxSync(tx []byte) (res types.Result)
-	QuerySync(types.RequestQuery) (resQuery types.ResponseQuery, err error)
-	CommitSync() (res types.Result)
+	EchoSync(msg string) (*types.ResponseEcho, error)
```

Comment:
these are breaking changes

Reply:
While we're at it, should we just make everything return (ResponseXxx, error) ? Seems like the best approach for future-proofing

Reply:
So, for example, App returns `ResponseDeliverTx{Code: CodeType_InternalError, Log: "bad"}` => abci client returns `nil, fmt.Errorf("%v: %s", r.Code, r.Log)` making interface more Golang style.. yeah, fine with me

Reply:
Not quite. Client returning an error should indicate an internal error, like issue with connecting to the app. App errors should be recorded as ABCI code and should not show up as client error. Does that make sense ?

Reply:
Ok, but we should make that clear in client docs